### PR TITLE
Support for collections 🗃️

### PR DESCRIPTION
**Pull request by nikagra**
Follow up to #21 which in turn is a continuation of #12 

## Support for Collections and UDTs in Scylla CDC Source Connector

This PR adds comprehensive support for ScyllaDB collection types (`LIST`, `SET`, `MAP`, `TUPLE`) and User Defined Types (`UDT`) to the Debezium-based CDC connector.

### Key Features

- **Collection and UDT Support**: 
  - Both frozen and non-frozen collections (`LIST`, `SET`, `MAP`) and UDT columns are now supported.
  - Frozen collections and UDTs are represented as single fields, preserving empty vs. `NULL` semantics.
  - Non-frozen collections are supported in "delta" mode, representing changes as a struct with `mode` and `elements` fields for granular tracking of modifications, overwrites, and removals.

- **New Configuration Option**:
  - Introduces `scylla.collections.mode` (enum `CollectionsMode`, default: `DELTA`) to control non-frozen collection representation.
  - Configuration is parsed and exposed via the connector config.

- **Schema and Translation Refactoring**:
  - Recursive schema computation and field translation logic to handle nested and complex types.
  - Refactored to use new `Field` class for improved type safety and clarity.

- **Enums for Clarity**:
  - New enums: `CollectionOperation` and `CollectionsMode` for explicit handling of collection operations and configuration.

- **Documentation**:
  - `README.md` updated with detailed documentation and examples for both frozen and non-frozen collections, UDTs, and the new configuration option.
  - Edge cases and known limitations (e.g., ambiguous empty vs. `NULL` for non-frozen collections on `INSERT`) are documented.

- **Configuration Defaults Cleanup**:
  - Cleaned up pooling and retry option defaults, removed unnecessary `.optional()` calls, and ensured all defaults are set.

- **Testing**:
  - Added and enhanced integration tests for frozen/non-frozen collections, UDTs, and edge cases.

### Known Limitations

- For non-frozen collections, Scylla CDC does not distinguish between explicit empty (`[]`/`{}`) and `NULL` on `INSERT`; the connector emits a top-level `null` in this ambiguous case.
- Only "delta" mode is currently supported for non-frozen collections.
- Partition/range deletes remain unsupported.

### Related Issues

- #201 

---

**Summary of changes:**  
- Adds support for all ScyllaDB collection and UDT types in CDC events.
- Introduces new configuration and enums for collection handling.
- Refactors schema and translation logic for recursive/nested types.
- Updates documentation and tests for new features and edge cases.

### CODE REVIEW

File: .github/copilot-instructions.md [hunk 1]
```diff
@@ -15,14 +15,26 @@ These instructions help AI coding agents quickly work productively in this repo.
...
+- `ScyllaCollectionSchema`, `CollectionsMode`, `CollectionOperation`: Schema-related classes/enums present in the codebase; non-frozen collections are currently supported only in **delta** mode as described in the README.
 - `transforms/ScyllaExtractNewState`: SMT flattening one-field Cell wrappers for easier sinks (similar to Debezium ExtractNewRecordState).
 
 ## Data Model & Topics
 - **Topic naming:** `logicalName.keyspace.table` from `scylla.name` (see README examples).
 - **Keys:** Primary key columns only.
 - **Values:** `op`, `before`, `after`, `source`, `ts_ms`; non-PK columns wrapped in single-field `Cell` structs to distinguish non-modification vs `NULL`.
 
+### Collections specifics
+- **Frozen collections:** Represented as a single Cell `value` field; empty vs `NULL` is preserved and asserted by `ScyllaTypesIT.canReplicateFrozenCollectionsEdgeCases`.
```
Comment (dkropachev) [inline .github/copilot-instructions.md]: Could you please detail what it means:
```
empty vs `NULL` is preserved and asserted by `ScyllaTypesIT.canReplicateFrozenCollectionsEdgeCases`.
```

File: src/main/java/com/scylladb/cdc/debezium/connector/ScyllaChangeRecordEmitter.java [hunk 2]
```diff
@@ -183,20 +194,31 @@ protected void emitDeleteRecord(
...
+            translateFieldToKafka(
+                change.getCell(cdef.getColumnName()),
+                schema.keySchema().field(cdef.getColumnName()).schema());
         valueStruct.put(cdef.getColumnName(), value);
-        if (keyStruct != null) {
-          keyStruct.put(cdef.getColumnName(), value);
-        }
+        keyStruct.put(cdef.getColumnName(), value);
       } else {
+        Schema cellSchema = schema.cellSchema(cdef.getColumnName());
```
[the remark this answers is not in the document]
Comment (nikagra) [reply src/main/java/com/scylladb/cdc/debezium/connector/ScyllaChangeRecordEmitter.java]: I think logging will be helpful there too

File: src/main/java/com/scylladb/cdc/debezium/connector/CollectionsMode.java [hunk 3]
```diff
@@ -0,0 +1,5 @@
+package com.scylladb.cdc.debezium.connector;
+
+public enum CollectionsMode {
+  DELTA,
```
[the remark this answers is not in the document]
Comment (nikagra) [reply src/main/java/com/scylladb/cdc/debezium/connector/CollectionsMode.java]: Here is my understanding. Pre-image concerns `"before"` field in produced message and delta mode concerns handling only collections: originally in `"after"`, after introduction of preimage in both `"before"` and `"after"` fields. The settings are orthogonal in a sense that they define different things: you can have preimage without collections support or collections support without preimage. But I agree that in the long run common `DELTA` setting is misleading since we may want to have different mods for different parts of the produced message (say, `full` in `before` and `delta` in `after`). 

Using this setting for now should be fine though: we only have delta and this is how collections represented

File: src/main/java/com/scylladb/cdc/debezium/connector/transforms/ScyllaExtractNewRecordState.java [hunk 4]
```diff
@@ -223,11 +273,30 @@ private Schema makeUpdatedSchema(Schema schema) {
         Schema innerSchema = field.schema().field("value").schema();
         if (isNonFrozenCollectionSchema(innerSchema)) {
           builder.field(field.name(), makeFlattenedNonFrozenSchema(innerSchema));
+        } else if (isMapSchema(innerSchema)) {
+          Schema mapEntrySchema =
+              SchemaBuilder.struct()
+                  .field("key", innerSchema.keySchema())
+                  .field("value", innerSchema.valueSchema())
+                  .build();
```
Comment (nikagra) [inline src/main/java/com/scylladb/cdc/debezium/connector/transforms/ScyllaExtractNewRecordState.java]: nitpick: Similar code is used many times through out this file, maybe some helper method can make it DRY
